workflow_obs.py

```diff
@@ -420,6 +440,86 @@ def clean_for_zarr(ds: xr.Dataset) -> xr.Dataset:
                 path=CONFIG['paths']['task']
             )
 
+    # --- COHERENCE ---
+    if "coherence" in CONFIG["tasks"]:
+        logger.info("Started coherence task")
+        for search_param_dict in CONFIG["coherence"]["inputs"]:
+            logger.info(f"Checking: {search_param_dict}")
+            ds_dict = pcat.search(**search_param_dict).to_dataset_dict()
+
+            # Get the variables common to all datasets
+            var_lists = {name: set(ds.data_vars) for name, ds in ds_dict.items()}
+            shared_vars = set.intersection(*var_lists.values())
+
+            # Construct source and variable strings
+            sources = '-'.join(sorted(ds.attrs["cat:source"] for ds in ds_dict.values()))
+            variable_string = '-'.join(sorted(shared_vars))
```

Comment:
why are you joining the variable ? why not iter on variables and one file per var ?